data-prepper-api/src/test/java/org/opensearch/dataprepper/model/buffer/AbstractBufferTest.java

```diff
@@ -198,6 +212,22 @@ public void testWriteAllTimeoutMetric() throws TimeoutException {
         Assert.assertEquals(1.0, timeoutMeasurements.get(0).getValue(), 0);
     }
 
+    @Test
+    public void testWriteAllRecordsWriteFailedMetric() {
+        // Given
+        final AbstractBuffer<Record<String>> abstractBuffer = new AbstractBufferRuntimeExceptionImpl(testPluginSetting);
+        final Collection<Record<String>> testRecords = Arrays.asList(
+                new Record<>(UUID.randomUUID().toString()), new Record<>(UUID.randomUUID().toString()));
+
+        // When/Then
+        Assert.assertThrows(RuntimeException.class, () -> abstractBuffer.writeAll(testRecords, 1000));
+
+        final List<Measurement> recordsWriteFailedMeasurements = MetricsTestUtil.getMeasurementList(
+                new StringJoiner(MetricNames.DELIMITER).add(PIPELINE_NAME).add(BUFFER_NAME).add(MetricNames.RECORDS_WRITE_FAILED).toString());
+        Assert.assertEquals(1, recordsWriteFailedMeasurements.size());
+        Assert.assertEquals(2.0, recordsWriteFailedMeasurements.get(0).getValue(), 0);
```

Comment:
QQ. What's the thrid parameter `0` here in assertEquals?

Reply:
I do find Hamcrest helpful for readability in such scenarios.

```
assertThat(recordsWriteFailedMeasurements.get(0).getValue(), closeTo(2.0, 0));
```